src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -7941,113 +7947,153 @@ instruct xorL_reg_imm(iRegLNoSp dst, iRegL src1, immLAdd src2) %{
 // ============================================================================
 // MemBar Instruction
 
-instruct load_fence() %{
+// RVTSO
+
+instruct unnecessary_membar_rvtso() %{
+  predicate(UseZtso);
   match(LoadFence);
-  ins_cost(ALU_COST);
+  match(StoreFence);
+  match(StoreStoreFence);
+  match(MemBarAcquire);
+  match(MemBarRelease);
+  match(MemBarStoreStore);
+  match(MemBarAcquireLock);
+  match(MemBarReleaseLock);
 
-  format %{ "#@load_fence" %}
+  ins_cost(0);
+
+  size(0);
 
+  format %{ "unnecessary_membar_rvtso elided/tso (empty encoding)" %}
   ins_encode %{
-    __ membar(MacroAssembler::LoadLoad | MacroAssembler::LoadStore);
+    __ block_comment("unnecessary_membar_rvtso");
   %}
-  ins_pipe(pipe_serial);
+  ins_pipe(real_empty);
 %}
 
-instruct membar_acquire() %{
-  match(MemBarAcquire);
-  ins_cost(ALU_COST);
+instruct membar_volatile_rvtso() %{
+  predicate(UseZtso);
+  match(MemBarVolatile);
+  ins_cost(VOLATILE_REF_COST);
 
-  format %{ "#@membar_acquire\n\t"
-            "fence ir iorw" %}
+  format %{ "membar_volatile_rvtso\n\t"
```

Comment:
It's a bit confusing to me to see this `membar_volatile_rvtso` followed by `fence w, r`.
Can you put a `#@` prefix making it a code comment for this `fence w, r`? I mean: `#@membar_volatile_rvtso`

Reply:
No other platform uses '@' in their ad files?
E.g.
```
  format %{ "membar_acquire\n\t"
                     "dmb ishld" %}
```

It just looked weird that riscv have it's special way to format things.

I can agree with '#', but why add a @ ?

Reply:
> I can agree with '#', but why add a @ ?

I can't recall the history. Maybe just to mark that this is the start of a specific match rule thus making the opto asm more readable. Output of some of the match rules are kind of complex like the compare ones (`CmpF3` etc.). But a single `#` also works for me.

Reply:
Ok, reverted to original style for now.